pkg/network/setup/network_test.go

```diff
@@ -288,6 +300,77 @@ var _ = Describe("VMNetworkConfigurator", func() {
 			})
 		})
 	})
+
+	Context("pod network phase1", func() {
+		var (
+			mockNetworkH *netdriver.MockNetworkHandler
+
+			vmi                   *v1.VirtualMachineInstance
+			vmNetworkConfigurator *VMNetworkConfigurator
+		)
+
+		BeforeEach(func() {
+			ctrl := gomock.NewController(GinkgoT())
+			mockNetworkH = netdriver.NewMockNetworkHandler(ctrl)
+
+			dutils.MockDefaultOwnershipManager()
+
+			vmi = newVMIBridgeInterface("testnamespace", "testVmName")
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+			vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+			vmNetworkConfigurator = newVMNetworkConfiguratorWithHandlerAndCache(vmi, mockNetworkH, &baseCacheCreator)
+		})
+
+		It("fails setup during network discovery", func() {
+			mockNetworkH.EXPECT().ReadIPAddressesFromLink(gomock.Any()).Return("", "", fmt.Errorf("discovery error"))
+
+			err := vmNetworkConfigurator.SetupPodNetworkPhase1(0, vmi.Spec.Networks)
+			var errCritical *neterrors.CriticalNetworkError
+			Expect(errors.As(err, &errCritical)).To(BeFalse(), "expected a non-critical error, but got %v", err)
```

Comment:
I believe the test will succeed even if no error was returned please add and `Expect` verify the error is not `nil`.